src/generic/trilinos_eigen_solver.h

```diff
@@ -765,6 +854,116 @@ namespace oomph
+                                              DoubleMultiVectorOperator>>
+        anasazi_pt = Teuchos::rcp(
+          new Anasazi::BasicEigenproblem<double,
+                                         DoubleMultiVector,
+                                         DoubleMultiVectorOperator>(Op_pt,
+                                                                    initial));
+
+      // Think I have it?
+      anasazi_pt->setHermitian(false);
+
+      // set the number of eigenvalues requested
+      anasazi_pt->setNEV(n_eval);
+
+      // Inform the eigenproblem that you are done passing it information
+      if (anasazi_pt->setProblem() != true)
+      {
+        oomph_info << "Anasazi::BasicEigenproblem::setProblem() had an error."
+                   << std::endl
+                   << "End Result: TEST FAILED" << std::endl;
+      }
+
+      // Create the solver manager
+      // No need to have ncv specificed, Triliinos has a sensible default
+      //  int ncv = 10;
+      MT tol = 1.0e-10;
+      int verbosity =
+        Anasazi::Warnings + Anasazi::FinalSummary + Anasazi::TimingDetails;
+
+      Teuchos::ParameterList MyPL;
+      MyPL.set("Which", "LM");
+      MyPL.set("Block Size", 1);
+      //  MyPL.set( "Num Blocks", ncv );
+      MyPL.set("Maximum Restarts", 500);
+      MyPL.set("Orthogonalization", "DGKS");
+      MyPL.set("Verbosity", verbosity);
+      MyPL.set("Convergence Tolerance", tol);
+      Anasazi::BlockKrylovSchurSolMgr<double,
+                                      DoubleMultiVector,
+                                      DoubleMultiVectorOperator>
+        BKS(anasazi_pt, MyPL);
+
+      // Solve the problem to the specified tolerances or length
+      Anasazi::ReturnType ret = BKS.solve();
+      bool testFailed = false;
+      if (ret != Anasazi::Converged)
+      {
+        testFailed = true;
+      }
+
+      if (testFailed)
+      {
+        oomph_info << "Eigensolver not converged\n";
+      }
+
+      // Get the eigenvalues and eigenvectors from the eigenproblem
+      Anasazi::Eigensolution<double, DoubleMultiVector> sol =
+        anasazi_pt->getSolution();
+      std::vector<Anasazi::Value<double>> evals = sol.Evals;
+      Teuchos::RCP<DoubleMultiVector> evecs = sol.Evecs;
+
+      eigenvalue.resize(evals.size());
+      eigenvector.resize(evals.size());
+
+      for (unsigned i = 0; i < evals.size(); i++)
+      {
+        // Undo shift and invert
+        double a = evals[i].realpart;
+        double b = evals[i].imagpart;
+        double det = a * a + b * b;
+        eigenvalue[i] = std::complex<double>(a / det + Sigma, -b / det);
+
+        // Now set the eigenvectors, I hope
+        eigenvector[i].build(evecs->distribution_pt());
+        unsigned nrow_local = evecs->nrow_local();
+        // Would be faster with pointers, but I'll sort that out later!
+        for (unsigned n = 0; n < nrow_local; n++)
+        {
+          eigenvector[i][n] = (*evecs)(i, n);
+        }
```

Comment:
Out of curiosity, is this your comment or something left over from Jack's work? If it's the latter, it's probably setting this as a `TODO` in the comment so that someone else can come along and fix it in the future

Reply:
Don't recommend using TODO in the code. If we don't understand what's going on, let's just delete the comment (or rather replace it with something descriptive -- can't see anything wrong with that.

Reply:
Sounds good to me!

Reply:
Me too. I think it was my comment, but it's way, way down the todo list.

Reply:
I've removed this comment. The previous comment provides enough description for this block.